src/main/java/atoml/data/DataDescription.java

```diff
@@ -174,6 +181,67 @@ public DataDescription(String name, int numFeatures, int numInformative, int num
 		
 		this.type = DescriptionType.GENERATED;
 	}
+	
+	/**
+	 * Creates a new data description for data that is generated
+	 *
+	 * @param name
+	 *            name of the data
+	 * @param numFeatures
+	 *            number of features
+	 * @param numInformative
+	 *            number of informative features, i.e., that are used to define the
+	 *            class
+	 * @param numInstances
+	 *            number of instances
+	 * @param distribution
+	 *            distribution of the numeric features
+	 * @param noiseRate
+	 *            noise rate
+	 * @param featureTypes
+	 *            types of the features; values>0 mean categorical features, the
+	 *            value defines the number of categories
+	 */
+	public DataDescription(String name, int numFeatures, int numInformative, int numInstances,
```

Comment:
Ideally try to have only one constructor with logic that is called by the others with default values. In this case, this should be this constructor that is called by the current constructors with distribution2=null. 

This removes code duplication and makes the class easier to maintain.